### PR TITLE
feat(neon_framework): Improved server icons

### PR DESCRIPTION
**Pull request by devesh-2002**
Issue #2003, 
I have tried to parse the css file and extract icons from there.
Please let me know, if there are any changes needed.

### CODE REVIEW

File: tool/generate-assets.sh [hunk 1]
```diff
@@ -7,6 +7,28 @@ color="#f37736"
...
+    icon_name="$1"
+    icon_svg_url=$(grep "$icon_name" /tmp/icon_mapping.txt | awk '{print $2}')
+    echo "$icon_svg_url"
+}
+
+function download_icon_svg() {
+    icon_name="$1"
+    icon_svg_url=$(get_svg_url "$icon_name")
+    wget "$icon_svg_url" -O "/tmp/$icon_name.svg"
+}
```
Comment (provokateurin) [inline tool/generate-assets.sh:29]: Unused

File: tool/generate-assets.sh [hunk 2]
```diff
@@ -7,6 +7,28 @@ color="#f37736"
 rm -rf /tmp/nextcloud-neon
 mkdir -p /tmp/nextcloud-neon
 
+function parse_icons_css() {
+    icons_css_url="https://raw.githubusercontent.com/nextcloud/server/master/dist/icons.css"
```
Comment (provokateurin) [inline tool/generate-assets.sh:11]: This file should be read from the git submodule at external/nextcloud-server instead

**Review by provokateurin — CHANGES_REQUESTED** [review]
The new code should replace the behavior of https://github.com/devesh-2002/neon/blob/843306c3ddf34e00ba60eb204703db2d6587900d/tool/generate-assets.sh#L88-L94, but instead of just reading the files from the paths it should extract the svgs contained in the css file as the icon files with their respective names.

File: tool/generate-assets.sh (hunk 2 above)
Comment (devesh-2002) [reply tool/generate-assets.sh:11]: How would I do that?

**Comment by provokateurin** [comment]
Given my comments in https://github.com/nextcloud/neon/issues/2003 this PR should probably be closed as this problem is a lot harder than I assumed and it's not really clear that the effort is even necessary as there are currently no known problems with the icons. @devesh-2002 would you be fine with that?

**Comment by devesh-2002** [comment]
Yes, You can go ahead and close this PR. Thanks a lot for all the help!

**Pull request closed without merging**